crates/router/src/core/webhooks.rs

```diff
@@ -478,9 +479,33 @@ pub async fn create_event_and_trigger_outgoing_webhook<W: api::OutgoingWebhookTy
             timestamp: event.created_at,
         };
 
+        let webhook_signature_payload =
+            ext_traits::Encode::<serde_json::Value>::encode_to_string_of_json(&outgoing_webhook)
+                .change_context(errors::WebhooksFlowError::OutgoingWebhookEncodingFailed)?;
+
+        let outgoing_webhooks_signature = merchant_account
+            .payment_response_hash_key
+            .clone()
+            .map(|key| {
+                common_utils::crypto::HmacSha512::sign_message(
+                    &common_utils::crypto::HmacSha512,
+                    key.as_bytes(),
+                    webhook_signature_payload.as_bytes(),
+                )
+            })
+            .transpose()
+            .change_context(errors::WebhooksFlowError::MerchantConfigNotFound)
```

Comment:
Signing failure is not a `MerchantConfigNotFound` error. Create a new error variant for this case in `WebhooksFlowError`.